tests/backup_restore_test.go

```diff
@@ -109,131 +109,169 @@ func TestBackupRestoreBasic(t *testing.T) {
 // 2. insert more data, and this is the point-in-time (denoted as t1) that we want to recover
 // 3. keep inserting data into the original tables
 // 4.1. set foreign_key_checks=OFF
-// 4.2. restore full backup at t0 to ghost tables
-// 4.3. apply binlog from t0 to t1 to ghost tables
+// 4.2. restore full backup at t0 to pitr tables
+// 4.3. apply binlog from t0 to t1 to pitr tables
 // 4.4. foreign_key_checks=ON
-// 5. lock tables and atomically swap original and ghost tables
+// 5. lock tables and atomically swap original and pitr tables
 func TestPITR(t *testing.T) {
 	t.Parallel()
 	a := require.New(t)
 
-	localhost := "127.0.0.1"
+	// common configs
+	const (
+		localhost    = "127.0.0.1"
+		username     = "root"
+		database     = "backup_restore"
+		numRowsTime0 = 10
+		numRowsTime1 = 20
+	)
 	port := getTestPort(t.Name())
-	username := "root"
-	database := "backup_restore"
 
-	insertData := func(db *sql.DB, a *require.Assertions, begin, end int) {
+	// common PITR routines
+	initDB := func(t *testing.T) (*sql.DB, func()) {
+		a := require.New(t)
+
+		_, stopFn := mysql.SetupTestInstance(t, port)
+
+		db, err := sql.Open("mysql", fmt.Sprintf("%s@tcp(%s:%d)/mysql", username, localhost, port))
+		a.NoError(err)
+
+		_, err = db.Exec(fmt.Sprintf("CREATE DATABASE %s", database))
```

Comment:
Move statements from line139 to 165 to a constant string and execute only once?

Reply:
It seems that `database/sql` does not support multiple statements in one SQL.